kuka_eki_hw_interface/src/kuka_eki_hw_interface.cpp

```diff
@@ -269,7 +287,16 @@ void KukaEkiHardwareInterface::read(const ros::Time &time, const ros::Duration &
 
 void KukaEkiHardwareInterface::write(const ros::Time &time, const ros::Duration &period)
 {
-  eki_write_command(joint_position_command_);
+  // only write if max will not be exceeded
+  if (eki_cmd_buff_len_ < eki_max_cmd_buff_len_)
+    eki_write_command(joint_position_command_);
+
+  // underflow/overflow checking
+  //if (eki_cmd_buff_len_ >= eki_max_cmd_buff_len_)
+  //  ROS_WARN_STREAM("eki_hw_iface RobotCommand buffer overflow (curent size " << eki_cmd_buff_len_
+  //                  << " greater than or equal max allowed " << eki_max_cmd_buff_len_ << ")");
+  //else if (eki_cmd_buff_len_ == 0)
+  //  ROS_WARN_STREAM("eki_hw_iface RobotCommand buffer empty");
```

Comment:
Is this commented for a reason? Doesn't it work or does it have unwanted side effects?

Reply:
Yes, this works it just results in a lot of console messages.  What about a debugging parameter or the like that enables/disables reporting?

Reply:
You could use `ROS_DEBUG_STREAM(..)` if we'd want to keep this in.

But all logging breaks determinism, so perhaps that would not be a good thing to add.